src/SFML/Window/Win32/WindowImplWin32.cpp

```diff
@@ -1174,6 +1253,35 @@ LRESULT CALLBACK WindowImplWin32::globalOnEvent(HWND handle, UINT message, WPARA
     return DefWindowProcW(handle, message, wParam, lParam);
 }
 
+////////////////////////////////////////////////////////////
+void WindowImplWin32::prepareTouch()
+{
+    static bool prepared = false;
+    if (!prepared)
+    {
+        prepared = true;
+
+        // Reset touch IDs
+        for (int i = 0; i < 10; ++i)
+            touchIDs[i] = static_cast<DWORD>(-1);
+    }
+
+    if (m_handle)
+        RegisterTouchWindow(m_handle, 0);
```

Comment:
What if `m_handle` isn't set? The rest of this function would still be executed, is that intended?

Can this even happen, after we [initialize it here](https://github.com/SFML/SFML/pull/912/files#diff-45590846459f313d131a3c29b7204fbecbed723a9d6c6d9138970d008b53383bR224-R234)? There is another `if (m_handle)` check that has been there before your changes, is that meaningful at all?

Reply:
Yeah, not sure, but looking at it, it feels even more weird, if you consider lines 241 and below. Without a proper handle, none of the other statements would make any sense anyway and you shouldn't increase window count, etc.